src/js/utils/strings.js

```diff
@@ -99,6 +99,19 @@ export function seconds(str, frameRate) {
     return sec;
 }
 
+// Convert an offset string to a number; supports conversion of percentage offsets
+export function offsetSeconds(offset, duration, frameRate) {
+    if (_.isString(offset) && offset.slice(-1) === '%') {
+        const percent = parseFloat(offset);
+        if (duration && !isNaN(percent)) {
+            return duration * percent / 100;
+        }
+        return 0;
+    } else if (!isNaN(Number(offset))) {
+        return Number(offset);
```

Comment:
Can we use `parseInt` or `parseFloat` instead? The `Number` constructor can cast all sorts of stuff, including hex, which we may not want.